### PR TITLE
Consolidate `InterruptCurrentForeground` and `MustRunInForeground`

### PR DESCRIPTION
**Pull request by andschwa**
And set them consistently. If the task needs to interrupt or run in the foreground, it really needs to do both. Also fixes some indentation.

### CODE REVIEW

**Comment by andschwa** [comment]
This was tested on top of #1778 and good to go @SeeminglyScience

**Review by SeeminglyScience — APPROVED** [review]
LGTM!

File: src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs [hunk 1]
```diff
@@ -25,9 +25,11 @@ internal class ConfigurationDoneHandler : IConfigurationDoneHandler
         // TODO: We currently set `WriteInputToHost` as true, which writes our debugged commands'
         // `GetInvocationText` and that reveals some obscure implementation details we should
         // instead hide from the user with pretty strings (or perhaps not write out at all).
+        //
+        // This API is mostly used for F5 execution so it requires the foreground.
         private static readonly PowerShellExecutionOptions s_debuggerExecutionOptions = new()
         {
-            MustRunInForeground = true,
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs:30]: @SeeminglyScience just looking back at this PR, knowing what we know now, we've moved F5 to priority next (seems right to me!)

File: src/PowerShellEditorServices/Services/Extension/ExtensionService.cs [hunk 2]
```diff
@@ -129,17 +129,16 @@ public Task InvokeCommandAsync(string commandName, EditorContext editorContext,
...
                 return ExecutionService.ExecutePSCommandAsync(
                     executeCommand,
                     cancellationToken,
                     new PowerShellExecutionOptions
                     {
+                        RequiresForeground = true,
                         WriteOutputToHost = !editorCommand.SuppressOutput,
                         AddToHistory = !editorCommand.SuppressOutput,
                         ThrowOnError = false,
-                        InterruptCurrentForeground = true
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/Extension/ExtensionService.cs:142]: Editor command execution was already going to run with priority next.

File: src/PowerShellEditorServices/Services/PowerShell/Console/PsrlReadLine.cs [hunk 3]
```diff
@@ -32,7 +32,11 @@ internal class PsrlReadLine : TerminalReadLine
             _psrlProxy.OverrideIdleHandler(onIdleAction);
         }
 
-        public override string ReadLine(CancellationToken cancellationToken) => _psesHost.InvokeDelegate(representation: "ReadLine", new ExecutionOptions { MustRunInForeground = true }, InvokePSReadLine, cancellationToken);
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/PowerShell/Console/PsrlReadLine.cs:35]: ReadLine will now run with priority next (again, seems right!)

File: src/PowerShellEditorServices/Services/PowerShell/Handlers/EvaluateHandler.cs [hunk 4]
```diff
@@ -21,18 +21,17 @@ internal class EvaluateHandler : IEvaluateHandler
...
                 new PSCommand().AddScript(request.Expression),
                 CancellationToken.None,
                 new PowerShellExecutionOptions
                 {
+                    RequiresForeground = true,
                     WriteInputToHost = true,
                     WriteOutputToHost = true,
                     AddToHistory = true,
                     ThrowOnError = false,
-                    InterruptCurrentForeground = true
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/PowerShell/Handlers/EvaluateHandler.cs:35]: F8 was already going to run with priority next, which F5 now matches.

File: src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs [hunk 5]
```diff
@@ -1146,7 +1160,7 @@ private Task PopOrReinitializeRunspaceAsync()
             // we simply run this on its thread, guaranteeing that no other action can occur
             return ExecuteDelegateAsync(
                 nameof(PopOrReinitializeRunspaceAsync),
-                new ExecutionOptions { InterruptCurrentForeground = true },
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs:1149]: PopOrReinitialize w as already going to run with priority next.

File: src/PowerShellEditorServices/Services/Template/TemplateService.cs [hunk 6]
```diff
@@ -158,15 +158,21 @@ public async Task<bool> ImportPlasterIfInstalledAsync()
...
+            PSCommand command = new PSCommand()
+                .AddCommand("Invoke-Plaster")
+                .AddParameter("TemplatePath", templatePath)
+                .AddParameter("DestinationPath", destinationPath);
 
+            // This command is interactive so it requires the foreground.
             await _executionService.ExecutePSCommandAsync(
                 command,
                 CancellationToken.None,
-                new PowerShellExecutionOptions { WriteOutputToHost = true, InterruptCurrentForeground = true, ThrowOnError = false }).ConfigureAwait(false);
```
Comment (andschwa) [inline src/PowerShellEditorServices/Services/Template/TemplateService.cs:169]: As was the Invoke-Plaster.

**Pull request merged**